File: backend/app/models/country.py
from sqlalchemy import Column, Integer, String, Boolean, DECIMAL, DateTime, func
from sqlalchemy.dialects.postgresql import JSONB
from app.database import Base


class Country(Base):
    __tablename__ = "countries"

    id = Column(Integer, primary_key=True, autoincrement=True)
    code = Column(String(3), unique=True, nullable=False)
    name_en = Column(String(100), nullable=False)
    name_ar = Column(String(100), nullable=False)
    currency_code = Column(String(3), nullable=False)
    currency_symbol_en = Column(String(10), nullable=False)
    currency_symbol_ar = Column(String(10), nullable=False)
    vat_rate = Column(DECIMAL(5, 2), default=0.00)
    payment_gateway = Column(String(20), nullable=False, default="placeholder")
    sms_provider = Column(String(20), nullable=False, default="twilio")
    phone_prefix = Column(String(5), nullable=False)
    default_lang = Column(String(2), default="ar")
    is_active = Column(Boolean, nullable=False, default=False)
    launched_at = Column(DateTime(timezone=True), nullable=True)
    config = Column(JSONB, default={})
    created_at = Column(DateTime(timezone=True), server_default=func.now(), nullable=False)

File: flutter_app/lib/models/country.dart
import 'package:json_annotation/json_annotation.dart';

part 'country.g.dart';

@JsonSerializable()
class Country {
  final int id;
  final String code;
  @JsonKey(name: 'name_en')
  final String nameEn;
  @JsonKey(name: 'name_ar')
  final String nameAr;
  @JsonKey(name: 'currency_code')
  final String currencyCode;
  @JsonKey(name: 'currency_symbol_en')
  final String currencySymbolEn;
  @JsonKey(name: 'currency_symbol_ar')
  final String currencySymbolAr;
  @JsonKey(name: 'vat_rate')
  final double vatRate;
  @JsonKey(name: 'phone_prefix')
  final String phonePrefix;
  @JsonKey(name: 'default_lang')
  final String defaultLang;
  @JsonKey(name: 'is_active')
  final bool isActive;

  const Country({
    required this.id,
    required this.code,
    required this.nameEn,
    required this.nameAr,
    required this.currencyCode,
    required this.currencySymbolEn,
    required this.currencySymbolAr,
    this.vatRate = 0,
    required this.phonePrefix,
    this.defaultLang = 'ar',
    this.isActive = false,
  });

  String name(String lang) => lang == 'ar' ? nameAr : nameEn;
  String currencySymbol(String lang) =>
      lang == 'ar' ? currencySymbolAr : currencySymbolEn;

  factory Country.fromJson(Map<String, dynamic> json) =>
      _$CountryFromJson(json);
  Map<String, dynamic> toJson() => _$CountryToJson(this);
}
